src/frontend/apps/impress/src/features/docs/doc-tree/components/DocSubPageItem.tsx

```diff
@@ -68,23 +80,49 @@ export const DocSubPageItem = (props: TreeViewNodeProps<Doc>) => {
     }
   };
 
+  useTreeItemKeyboardActivate(isActive, handleActivate);
+
+  // prepare the text for the screen reader
+  const docTitle = doc.title || untitledDocument;
+  const hasChildren = (doc.children?.length || 0) > 0;
+  const isExpanded = node.isOpen;
+  const isSelected = treeContext?.treeData.selectedNode?.id === doc.id;
+
+  const ariaLabel = `${docTitle}${hasChildren ? `, ${isExpanded ? t('expanded') : t('collapsed')}` : ''}${isSelected ? `, ${t('selected')}` : ''}`;
+
   return (
     <Box
       className="--docs-sub-page-item"
       draggable={doc.abilities.move && isDesktop}
       $position="relative"
+      role="treeitem"
+      aria-label={ariaLabel}
+      aria-selected={isSelected}
+      aria-expanded={hasChildren ? isExpanded : undefined}
       $css={css`
-        background-color: ${actionsOpen
+        background-color: ${isActive
           ? 'var(--c--theme--colors--greyscale-100)'
           : 'var(--c--theme--colors--greyscale-000)'};
 
         .light-doc-item-actions {
-          display: ${actionsOpen || !isDesktop ? 'flex' : 'none'};
+          display: flex;
+          opacity: ${isActive || !isDesktop ? 1 : 0};
+          visibility: ${isActive || !isDesktop ? 'visible' : 'hidden'};
```

Comment:
There is some blinking when I open the menu and close it.
https://github.com/user-attachments/assets/f345ffb9-23cb-40a3-9603-ae9522ceff26